bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/discovery/AndroidDebugBridgeDiscoveryService.java

```diff
@@ -138,8 +142,15 @@ private void discoverWithADB(String ip, int port) throws InterruptedException, A
             String model = device.getModel();
             String androidVersion = device.getAndroidVersion();
             String brand = device.getBrand();
-            logger.debug("discovered: {} - {} - {} - {}", model, serialNo, androidVersion, brand);
-            onDiscoverResult(serialNo, ip, port, model, androidVersion, brand);
+            String macAddress;
+            try {
+                macAddress = device.getMacAddress();
```

Comment:
I think I can replace this call to retrieve the Mac address from the device by accessing the hardware address of the `NetworkInterface` used to discover the device.